src/backend/input.rs

```diff
@@ -77,7 +77,17 @@ pub trait Event {
     fn time(&self) -> u32;
 }
 
-impl Event for () {
+/// Struct to mark events never emitted by an `InputBackend` implementation.
+///
+/// Implements all event types and can be used in place for any `Event` type,
+/// that is not used by an `InputBackend` implementation. Initialization is not
+/// possible, making accidential use impossible and enabling a lot of possible
+/// compiler optimizations.
+pub struct UnusedEvent {
+    _hidden_field: (),
+}
```

Comment:
Mostly it makes everything lighter by not pulling in the whole panic machinery for a panic that is in a dead code-path anyway.